### PR TITLE
feat: add inference job filter

### PR DESCRIPTION
**Pull request by o0OooO**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

截屏2023-08-07 20 45 50

### CODE REVIEW

**Review by alvindaiyan — CHANGES_REQUESTED** [review]

File: middleware_api/lambda/inference/app.py [hunk 1]
```diff
@@ -609,12 +708,14 @@ async def run_model_merge(request: Request):
...
-        current_time = get_curent_time()
+        current_time = str(datetime.now())
         response = inference_table.put_item(
             Item={
                 'InferenceJobId': inference_id,
                 'startTime': current_time,
-                'status': 'inprogress'
+                'status': 'inprogress',
+                'endpoint': endpoint_name,
+                'checkpoint': checkpoint_name,
```
Comment (alvindaiyan) [inline middleware_api/lambda/inference/app.py:718]: the new txt2img and img2img process is not using these code

**Review by alvindaiyan — CHANGES_REQUESTED** [review]

File: aws_extension/sagemaker_ui.py [hunk 2]
```diff
@@ -63,11 +63,48 @@
...
+img_task_type = None
+img_status = None
+img_endpoint = None
+img_checkpoint = None
+
+task_type_dropdown = None
+status_dropdown = None
+sagemaker_endpoint_filter = None
+sd_checkpoint_filter = None
+
```
[the remark this answers is not in the document]
Comment (o0OooO) [reply aws_extension/sagemaker_ui.py:85]: 不是global的设置值不生效

File: middleware_api/lambda/inference/app.py (hunk 1 above)
Comment (o0OooO) [reply middleware_api/lambda/inference/app.py:718]: 好的 我删掉～

Comment (alvindaiyan) [reply middleware_api/lambda/inference/app.py:718]: no delete pls, the api is still in use

**Comment by yike5460** [comment]
@o0OooO It's a good practice to squash or clean up commits with 'test with log' before opening a pull request to maintain a clean commit history.

**Review by alvindaiyan — APPROVED** [review]
details need to be tested more

**Review by alvindaiyan — CHANGES_REQUESTED** [review]
the code will not work with new inference api yet

File: aws_extension/sagemaker_ui.py [hunk 3]
```diff
@@ -847,18 +1021,128 @@ def create_ui(is_img2img):
...
-                interrogate_type_checkbox.change(update_txt2img_inference_job_ids, inputs=[inference_job_dropdown, txt2img_type_checkbox, img2img_type_checkbox, interrogate_type_checkbox], outputs=inference_job_dropdown)
+                        task_type_choices = ["txt2img", "img2img", "interrogate_clip", "interrogate_deepbooru"]
+                        task_type_dropdown = gr.Dropdown(label="Task Type", choices=task_type_choices, elem_id="task_type_ids_dropdown")
+                        status_choices = ["succeed", "inprogress", "failed"]
+                        status_dropdown = gr.Dropdown(label="Status", choices=status_choices, elem_id="task_status_dropdown")
+                    with gr.Row():
+                        sagemaker_endpoint_filter = gr.Dropdown(sagemaker_endpoints, label="SageMaker Endpoint", elem_id="sagemaker_endpoint_dropdown" )
+                        modules.ui.create_refresh_button(sagemaker_endpoint_filter, update_sagemaker_endpoints, lambda: {"choices": sagemaker_endpoints}, "refresh_sagemaker_endpoints")
```
Comment (alvindaiyan) [inline aws_extension/sagemaker_ui.py:1053]: can we empty the selector after we select one?

**Review by alvindaiyan — APPROVED** [review]

**Pull request merged**